src/lib/NextRouteHistory.js

```diff
@@ -26,7 +28,14 @@ class NextRouterHistory implements RouterHistory {
     const route = this.router.getRoute(name, true);
     const path = this.router.generate(name, params);
 
-    NextRouter[method]({ pathname: '/main', query: { routeName: route.name, ...params } }, path);
+    if (this.isCordova) {
+      const query = { routeName: route.name, path, ...params };
+      const pathname = window.location.pathname;
+      NextRouter[method]({ pathname: '/cordova', query }, { pathname, query });
```

Comment:
I think this solution of having two pages is okay. When we know that the implementation stays as similar as it is now after a while, we can merge the two implementations into one again – for now I'd keep both even if it means code duplication.